### PR TITLE
feat: method for reserving login for user

### PR DESCRIPTION
**Pull request by xflord**
* created method reserveLogin() in all layers
* it sets the user attribute loginNamespace in the selected namespace for the selected user, effectively reserving the login
* added the method to openapi

### CODE REVIEW

**Review by bodnara — COMMENTED** [review]
Looks good to me. Only question would be, if we want to write tests for this ?

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java [hunk 1]
```diff
@@ -1774,6 +1774,29 @@ public User call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	 * @throw InvalidLoginException When login to reserve has invalid syntax or is not allowed.
+	 * @throw AlreadyReservedLoginException When login is already reserved
+	 */
+	reserveLogin {
+		@Override
+		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+			parms.stateChangingCheck();
+			ac.getUsersManager().reserveLogin(ac.getSession(),
+				ac.getUserById(parms.readInt("user")),
+				parms.readString("userLogin"),
```
Comment (Johaney-s) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java:1795]: openapi's login is called just "login", so i think it wouldn't match to "userLogin" here, the same goes for "namespace" vs "loginNamespace"...

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/UsersManagerBl.java [hunk 2]
```diff
@@ -38,6 +38,7 @@
 import cz.metacentrum.perun.core.api.exceptions.PasswordResetLinkNotValidException;
 import cz.metacentrum.perun.core.api.exceptions.PasswordStrengthException;
 import cz.metacentrum.perun.core.api.exceptions.PasswordStrengthFailedException;
+import cz.metacentrum.perun.core.api.exceptions.PrivilegeException;
```
Comment (Johaney-s) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/UsersManagerBl.java:41]: unused import

**Review by zlamalp — COMMENTED** [review]
We already have similar API logic - method `setLogin()` which is used to set login for service accounts by their owners. I'm not sure about this method and its relation to VOADMIN role, we should probably discuss it.

**Pull request closed without merging**